source/static/js/download.js

```diff
@@ -0,0 +1,26 @@
+;
+(function(d, n) {
+  var os = n.platform.match(/(Win|Mac|Linux)/);
+  var x = n.userAgent.match(/x86_64|Win64|WOW64/) ||
+    n.cpuClass === 'x64' ? 'x64' : 'x86';
+  var base = 'https://iojs.org/dist/latest/';
+  var db = d.getElementById('home-downloadbutton');
+  var version = db.dataset.version;
+  var dlLocal = db.dataset.dlLocal;
+  var href;
+  switch (os && os[1]) {
+    case 'Mac':
+      href = 'iojs-' + version + '.pkg';
+      db.innerHTML = dlLocal + ' OS X (x64)';
+      break;
+    case 'Win':
+      href = 'iojs-' + version + '-' + x + '.msi';
+      db.innerHTML = dlLocal + ' Windows ('+x+')';
+      break;
+    case 'Linux':
+      href = 'iojs-' + version + '-linux-' + x + '.tar.gz';
+      db.innerHTML = dlLocal + ' Linux ('+x+')';
+      break;
+  }
+  db.href = base + href;
+})(document, navigator);
```

Comment:
- i am not a big fan of a download button based on JS - i know some people that use no-script.

 - the code also misses tests, we have that problem with the js-code on nodejs.org too, it makes changes very difficult, time-consuming and error prone.

- what happens with FreeBSD people or folks that use other platforms? from following the code path in my head (again, missing tests) it looks like they get: https://iojs.org/dist/latest/undefined 

- that also means whenever there is a platform change for binaries (new platform is supported, an old platform is not supported any more), we have to remember that we have hardcoded it in JS and if we remember it, touch this piece of code, having more maintenance in the end (or bugs if we don't remember)

- for node/io i am not a user of the website for downloads, I use nvm there. but i am a user of https://www.erlang-solutions.com/downloads which also has a download button and they sniff my user agent. it always makes me mad, as I usually want the address of their .deb repo for ubuntu, but I click the button, they sniff my user agent and send me to the OSX version, where I have to switch to ubuntu manually. for me that really feels like a bad ui as i have to click 1 more time to decide against the sniffed OS. i think a list with the three major versions on https://www.erlang-solutions.com/downloads would make the page more usable and I could imagine this also applies to node/io binaries. in the case for this button submitted in the PR the user would have to find the link thats sends them to the list of supported binaries after a download for a binary started which they can't make use of.